src/state.rs

```diff
@@ -84,16 +84,23 @@ impl ProgressState {
     }
 
     /// Return the current average time per step
+    #[allow(dead_code)] // Keep for API compatibility. Should deprecate.
     pub fn avg_time_per_step(&self) -> Duration {
-        self.est.time_per_step()
+        secs_to_duration(self.est.seconds_per_step())
+    }
+
+    /// Return the current average time in seconds per step.
```

Comment:
Style nit: please strip the period from the first comment line (it's a sentence fragment, not a sentence), and add an empty line between the two lines.